staging/src/k8s.io/kube-aggregator/pkg/controllers/openapi/controller.go

```diff
@@ -139,23 +227,40 @@ func (c *AggregationController) sync(key string) (syncAction, error) {
 // AddAPIService adds a new API Service to OpenAPI Aggregation.
 func (c *AggregationController) AddAPIService(handler http.Handler, apiService *apiregistration.APIService) {
 	if apiService.Spec.Service == nil {
+		// ignore local services
 		return
 	}
-	if err := c.openAPIAggregationManager.AddUpdateAPIService(handler, apiService); err != nil {
+
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	// TODO: combine APIServices from the same aggregated apiserver by choosing the same key
```

Comment:
This will be a follow-up PR. We have to update the auto registration controller. Currently it only looks at web services of the whole kube-apiserver and registers them. This loses the information about which delegate is responsible.